pkgs/native_assets_cli/lib/src/model/link_config.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of '../api/link_config.dart';
+
+/// The input to the linking script.
+///
+/// It consists of the fields inherited from the [HookConfig] and the [assets]
+/// from the build step.
+class LinkConfigImpl extends HookConfigImpl implements LinkConfig {
+  static const resourceIdentifierKey = 'resource_identifiers';
+
+  static const assetsKey = 'assets';
+
+  @override
+  final Iterable<AssetImpl> assets;
+
+  // TODO: Placeholder for the resources.json file URL. We don't want to change
+  // native_assets_builder when implementing the parsing.
+  final Uri? _resourceIdentifierUri;
+
+  LinkConfigImpl({
+    required this.assets,
+    Uri? resourceIdentifierUri,
```

Comment:
Let's remove this for this PR so that we don't accidentally pass something in Dart/Flutter.

Reply:
But then the API changes. Keeping it in keeps the API stable, and doesn't even cause breakages if you do accidentally pass something in, as it is ignored.

Reply:
(Also users can't use it, so there is no risk of breaking clients)

Reply:
Just to double check, it's never eagerly parsed, so it doesn't break if we change the serialization.

Reply:
Yes!